File: 이건희/level2/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)
    result = 0
    
    while True:
        first = heapq.heappop(scoville) 
        if first >= K:
            break
        if len(scoville) == 0:
            return -1
        second =  heapq.heappop(scoville)
        num = first + second * 2
        heapq.heappush(scoville, num)
        result += 1
        
    return result

File: 김찬우/프로그래머스 /Lv.2/더맵게.java
import java.util.*;


class Solution {
    public int solution(int[] scoville, int K) {
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int result = 0; 
        
        for(int i = 0; i<scoville.length; i++){
            pq.offer(scoville[i]);
        }
        
        
        while(pq.peek() < K){
            
            int first = pq.poll();
            
            if(pq.isEmpty()){
                result = -1;
                break;
            }
            
            int second = pq.poll();
            
            pq.offer(first + second * 2);
            
            result++;

            
        }

    
        return result;
    }
}
